sim/warrior/shield_block.go

```diff
@@ -7,60 +7,53 @@ import (
 	"github.com/wowsims/wotlk/sim/core/stats"
 )
 
-func (warrior *Warrior) registerShieldBlockSpell() {
+func (warrior *Warrior) RegisterShieldBlockCD() {
 	actionID := core.ActionID{SpellID: 2565}
+	blockValueMult := 2.0
 
 	shieldBlockAura := warrior.RegisterAura(core.Aura{
-		Label:     "Shield Block",
-		ActionID:  actionID,
-		Duration:  time.Second*5 + core.TernaryDuration(warrior.Talents.ImprovedShieldBlock, time.Second, 0),
-		MaxStacks: 2,
+		Label:    "Shield Block",
+		ActionID: actionID,
+		Duration: time.Second * 10,
 		OnGain: func(aura *core.Aura, sim *core.Simulation) {
-			warrior.AddStatDynamic(sim, stats.Block, 75*core.BlockRatingPerBlockChance)
+			warrior.AddStatDynamic(sim, stats.Block, 100*core.BlockRatingPerBlockChance)
+			// TODO: The innate block value from the shield item should not be multiplied
+			// as shown here https://youtu.be/LYJdkimJgn8?t=476
+			warrior.AddStatDependencyDynamic(sim, stats.BlockValue, stats.BlockValue, blockValueMult)
 		},
 		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
-			warrior.AddStatDynamic(sim, stats.Block, -75*core.BlockRatingPerBlockChance)
-		},
-		OnSpellHitTaken: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, spellEffect *core.SpellEffect) {
-			if spellEffect.Outcome.Matches(core.OutcomeBlock) {
-				aura.RemoveStack(sim)
-			}
+			warrior.AddStatDynamic(sim, stats.Block, -100*core.BlockRatingPerBlockChance)
+			warrior.AddStatDependencyDynamic(sim, stats.BlockValue, stats.BlockValue, 1.0/blockValueMult)
 		},
 	})
 
-	cost := 10.0
-	initialStacks := int32(1)
-	if warrior.Talents.ImprovedShieldBlock {
-		initialStacks++
-	}
+	cooldownDuration := time.Second * 60
+	cooldownDuration -= time.Second * 10 * time.Duration(warrior.Talents.ShieldMastery)
 
 	warrior.ShieldBlock = warrior.RegisterSpell(core.SpellConfig{
 		ActionID:    actionID,
 		SpellSchool: core.SpellSchoolPhysical,
 
-		ResourceType: stats.Rage,
-		BaseCost:     cost,
-
 		Cast: core.CastConfig{
-			DefaultCast: core.Cast{
-				Cost: cost,
-			},
+			DefaultCast: core.Cast{},
 			CD: core.Cooldown{
 				Timer:    warrior.NewTimer(),
-				Duration: time.Second * 5,
+				Duration: cooldownDuration,
```

Comment:
Nothing wrong with this, but it would be cleaner to instead write this as:

`Duration: time.Second*60 - time.Second*10*time.Duration(warrior.Talents.ShieldMastery),`

Just so the reader doesn't need to go searching to find the value.